web/src/app/schedules/temp-sched/TempSchedAddShiftsStep.tsx

```diff
@@ -131,52 +131,42 @@ export default function TempSchedAddShiftsStep({
   function fieldErrors(s = submitted): FieldError[] {
     const result: FieldError[] = []
 
-    if (!shift) {
+    if (!shift || !s) {
       return result
     }
 
-    if (s) {
-      const message = 'this field is required'
-      if (!shift.userID) {
-        result.push({
-          field: 'userID',
-          message,
-        } as FieldError)
-      }
-      if (!shift.start) {
-        result.push({
-          field: 'start',
-          message,
-        } as FieldError)
-      }
-      if (!shift.end) {
-        result.push({
-          field: 'end',
-          message,
-        } as FieldError)
-      }
-
-      return result
-    }
-
-    if (!isAfter(shift.end, shift?.start)) {
+    const message = 'this field is required'
+    if (!shift.userID) {
       result.push({
-        field: 'end',
-        message: 'must be after shift start time',
+        field: 'userID',
+        message,
       } as FieldError)
     }
-    if (isBefore(shift.start, start)) {
+    if (!shift.start) {
       result.push({
         field: 'start',
-        message: 'must not be before temporary schedule start time',
+        message,
+      } as FieldError)
+    }
+    if (!shift.end) {
+      result.push({
+        field: 'end',
+        message,
       } as FieldError)
     }
-    if (isAfter(shift.end, end)) {
+    if (!isAfter(shift.end, shift.start)) {
       result.push({
         field: 'end',
-        message: 'must not extend beyond temporary schedule end time',
+        message: 'must be after shift start time',
+      } as FieldError)
+    }
```

Comment:
While this is true, one is meant to display the error for the `start` field and the other for the `end` field

Reply:
If we want to display an error for both fields, that's fine, but as they are logically the exact same check (just written two ways) let's push both errors in the same if block to make the intention clear

Reply:
I'm not sure. I think I would prefer them separate, as it's the field the user is editing. If the user is editing `start` and not `end` and types a date past `end`, I think an error on `start` is sufficient because multiple fields flashing red is probably a bit more forward than we need to be.